src/sentry/api/event_search.py

```diff
@@ -442,16 +448,24 @@ def visit_has_filter(self, node, children):
             )
 
         operator = "=" if self.is_negated(negation) else "!="
-
         return SearchFilter(search_key, operator, SearchValue(""))
 
+    def visit_tag_filter(self, node, children):
+        (negation, _, search_key, _, sep, search_value) = children
+        operator = "!=" if self.is_negated(negation) else "="
+        return SearchFilter(SearchKey("tags[" + search_key.name + "]"), operator, search_value)
```

Comment:
You could have a unicode error here if someone does a search for `tags[öhnø]`.

Reply:
It doesn't appear to be throwing a unicode error at first test - but I'll look deeper and see what is being passed around internally in this case :) Will probably just change the concatenation method to be safe.